website/moneybirdsynchronization/signals.py

```diff
@@ -0,0 +1,359 @@
+"""The signals checked by the moneybrid synchronization package."""
+from django.db.models.signals import pre_save, post_save, pre_delete
+from django.template.defaultfilters import date
+import logging
+
+from utils.models.signals import suspendingreceiver
+
+from members.models import Member
+from moneybirdsynchronization.models import Contact
+from moneybirdsynchronization.administration import HttpsAdministration, Administration
+from moneybirdsynchronization import services
+from payments.models import Payment
+from sales.models.order import Order
+from events.models.event import Event
+from thaliawebsite import settings
+
+@suspendingreceiver(
+    post_save,
+    sender="members.Profile",
+)
+def post_profile_save(sender, instance, **kwargs):
+    """Update contact on profile creation."""
+    api = HttpsAdministration(settings.MONEYBIRD_API_KEY, settings.MONEYBIRD_ADMINISTRATION_ID)
+    member = Member.objects.get(profile=instance)
+    contact = Contact.objects.get_or_create(member=member)[0]
+    if contact.moneybird_version is None:
+        response = api.post("contacts", contact.to_moneybird())
+        contact.moneybird_id = response["id"]
+        contact.moneybird_version = response["version"]
+        contact.save()
+    else:
+        response = api.patch("contacts/{}".format(contact.moneybird_id), contact.to_moneybird())
+        contact.moneybird_version = response["version"]
+        contact.save()
+
+
+@suspendingreceiver(
+    pre_delete,
+    sender="members.Profile",
+)
+def post_profile_delete(sender, instance, **kwargs):
+    """Delete contact on profile deletion."""
+    api = HttpsAdministration(settings.MONEYBIRD_API_KEY, settings.MONEYBIRD_ADMINISTRATION_ID)
+    member = Member.objects.get(profile=instance)
+    contact = Contact.objects.get(member=member)
+    api.delete("contacts/{}".format(contact.moneybird_id))
+    contact.delete()
+
+
+@suspendingreceiver(
+    post_save,
+    sender="auth.User",
+)
+def post_user_save(sender, instance, **kwargs):
+    """Update contact on user creation."""
+    api = HttpsAdministration(settings.MONEYBIRD_API_KEY, settings.MONEYBIRD_ADMINISTRATION_ID)
+    contact = Contact.objects.get_or_create(member=instance)[0]
+    if contact.moneybird_version is not None:
+        response = api.patch("contacts/{}".format(contact.moneybird_id), contact.to_moneybird())
+        contact.moneybird_version = response["version"]
+        contact.save()
+
+
+@suspendingreceiver(
+    post_save,
+    sender="events.EventRegistration",
+)
+def post_event_registration_payment(sender, instance, **kwargs):
+    """ Create external invoice on payment creation."""
+    if instance.payment is None:
+        return
+    if instance.payment.moneybird_invoice_id is not None:
+        return
+    
+    api = HttpsAdministration(settings.MONEYBIRD_API_KEY, settings.MONEYBIRD_ADMINISTRATION_ID)
+    contact_id = api.get("contacts/customer_id/34")["id"]
```

Comment:
Is contact with customer id 34 the anonymous contact? That should be documented (and again, don't hardcode)